### DISCUSSION TITLE
Chance to not load beatmaps successfully on Linux

### DISCUSSION THREAD

**Discussion by manujedi** [ppy/osu#16179] [Q&A] [answered]
When i scroll through beatmaps there is a chance that the beatmap will not load successfully. It also happens when i try to start playing a beatmap. It feels like the chance for it to happen gets higher the longer osu is running. It happens with the appimage and if i run code directly(dotnet run...)

Error log:

```
[runtime] 2021-12-20 13:35:20 [verbose]: Song select working beatmap updated to Nightwish - Edema Ruh (EncoreW) [The Name of the Wind]
[runtime] 2021-12-20 13:35:20 [error]: Could not load beatmap successfully!
[runtime] 2021-12-20 13:35:20 [error]: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
[runtime] 2021-12-20 13:35:20 [error]: ---> System.ArgumentOutOfRangeException: Must be greater than 0. (Parameter 'Precision')
[runtime] 2021-12-20 13:35:20 [error]: Actual value was 0.
[runtime] 2021-12-20 13:35:20 [error]: at osu.Framework.Bindables.BindableNumber`1.set_Precision(T value)
[runtime] 2021-12-20 13:35:20 [error]: at osu.Game.Beatmaps.Formats.LegacyDecoder`1.LegacyDifficultyControlPoint..ctor() in /home/manu/tmp/osu/osu.Game/Beatmaps/Formats/LegacyDecoder.cs:line 180
[runtime] 2021-12-20 13:35:20 [error]: at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)
[runtime] 2021-12-20 13:35:20 [error]: --- End of inner exception stack trace ---
[runtime] 2021-12-20 13:35:20 [error]: at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)
[runtime] 2021-12-20 13:35:20 [error]: at osu.Game.Beatmaps.ControlPoints.ControlPoint.DeepClone() in /home/manu/tmp/osu/osu.Game/Beatmaps/ControlPoints/ControlPoint.cs:line 38
[runtime] 2021-12-20 13:35:20 [error]: at osu.Game.Rulesets.Objects.HitObject.ApplyDefaults(ControlPointInfo controlPointInfo, IBeatmapDifficultyInfo difficulty, CancellationToken cancellationToken) in /home/manu/tmp/osu/osu.Game/Rulesets/Objects/HitObject.cs:line 119
[runtime] 2021-12-20 13:35:20 [error]: at osu.Game.Beatmaps.WorkingBeatmap.GetPlayableBeatmap(IRulesetInfo ruleset, IReadOnlyList`1 mods, CancellationToken token) in /home/manu/tmp/osu/osu.Game/Beatmaps/WorkingBeatmap.cs:line 149
[runtime] 2021-12-20 13:35:20 [error]: at osu.Game.Beatmaps.WorkingBeatmap.GetPlayableBeatmap(IRulesetInfo ruleset, IReadOnlyList`1 mods) in /home/manu/tmp/osu/osu.Game/Beatmaps/WorkingBeatmap.cs:line 89
[runtime] 2021-12-20 13:35:20 [error]: at osu.Game.Screens.Select.BeatmapInfoWedge.WedgeInfoText.getRulesetInfoLabels() in /home/manu/tmp/osu/osu.Game/Screens/Select/BeatmapInfoWedge.cs:line 389
```

It is not the beatmap itself as reselcting the beatmap simply fixes it. If i can provide more logs just tell me where to find them/produce them. I don't have the problem on Windows.

**Comment by bdach** [comment, marked as answer]
The last time we've had a report like this it was pinpointed to the audio library we use (bass) messing with the FP control word. (Curious readers can refer to [this blog post of mine](https://bdach.github.io/debugging/2020/09/18/when-double-epsilon-can-equal-zero.html) for details).

I can't reproduce this on my box but similar steps could be taken on a linux box to check if it's this, except you'd have to compile a linux shared library rather than a dll like I did back then. I can't do this myself, because I don't have that issue.

**Reply by manujedi** [reply]
Yes that was it. After some more testing and replacing all occurrences of libbass.so 
```
./osu.Desktop/bin/Debug/net5.0/linux-x64/libbass.so
./osu.Desktop/bin/Debug/net5.0/runtimes/linux-x64/native/libbass.so
./osu.Desktop/bin/Release/net5.0/linux-x64/libbass.so
```
with this
[libbass.zip](https://github.com/ppy/osu/files/7747094/libbass.zip)
everything is working

**Reply by bdach** [reply]
Out of curiosity where did you get the "good" binaries you linked in the comment above?

**Reply by manujedi** [reply]
from www.un4seen.com/files/bass24-linux.zip
but i didn't test it more than an hour yesterday, but the bug normally happened very often. Chances are high that this fixed it.

**Reply by peppy** [reply]
It turns out we were running older releases than we expected. These have been updated for the next release, so you should see this fixed.
